tape/src/test/java/com/squareup/tape2/ObjectQueueTest.java

```diff
@@ -194,6 +194,28 @@
     }
   }
 
+  @Test public void iteratorThrowsIOException() throws IOException {
+    File parent = folder.getRoot();
+    File file = new File(parent, "object-queue");
+    QueueFile queueFile = new QueueFile.Builder(file).build();
+    ObjectQueue<Object> queue = ObjectQueue.create(queueFile, new ObjectQueue.Converter<Object>() {
+      @Override public String from(byte[] bytes) throws IOException {
+        throw new IOException();
+      }
+
+      @Override public void toStream(Object o, OutputStream bytes) {
+      }
+    });
+    queue.add(new Object());
+    Iterator<Object> iterator = queue.iterator();
+    try {
+      iterator.next();
+      fail();
+    } catch (Throwable ioe) {
+      assertThat(ioe).isInstanceOf(IOException.class);
```

Comment:
Just catch IOE? I'm surprised error-prone doesn't flag this pattern.

Reply:
agree this is a terrible pattern (because the fail assertion will fall in there).

i'm not sure i can catch Java's caught exceptions from a block that doesn't declare them. will have to check in a bit.

Reply:
oh, EP does catch this, it looks like. http://errorprone.info/bugpattern/AssertionFailureIgnored is failing CI here.

Reply:
i'll catch Exception here. then AssertionErrors won't be caught.